File: chatroom/client/lib/utils/models/contacts.dart
class Contact {
  int _id;
  String _name;
  String _surname;
  String _nickname;
  int _imageId;
  int _chatId;

  Contact(
      {int id,
      String name,
      String surname,
      String nickname,
      int imageId,
      int chatId}) {
    this._id = id;
    this._name = name;
    this._surname = surname;
    this._nickname = nickname;
    this._imageId = imageId;
    this._chatId = chatId;
  }

  int get id => _id;
  set id(int id) => _id = id;
  String get name => _name;
  set name(String name) => _name = name;
  String get surname => _surname;
  set surname(String surname) => _surname = surname;
  String get nickname => _nickname;
  set nickname(String nickname) => _nickname = nickname;
  int get imageId => _imageId;
  set imageId(int imageId) => _imageId = imageId;
  int get chatId => _chatId;
  set chatId(int chatId) => _chatId = chatId;

  Contact.fromJson(Map<String, dynamic> json) {
    _id = json['id'];
    _name = json['name'];
    _surname = json['surname'];
    _nickname = json['nickname'];
    _imageId = json['imageId'];
    _chatId = json['chatId'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this._id;
    data['name'] = this._name;
    data['surname'] = this._surname;
    data['nickname'] = this._nickname;
    data['imageId'] = this._imageId;
    data['chatId'] = this._chatId;
    return data;
  }
}

File: chatroom/server/modules/socket/data/contacts.js
const logger = require('../../logger');
const database = require('../../database');

module.exports = async (socket, data) => {
  if (true) { //TODO: verificare se il client ha inviato un TOKEN valido
    let dataToSand = [];
    let userId = database.escape(data.id);
    //TODO: inviare insieme all'utente l'id della chat, se esiste
    let users = await database.query('SELECT id, name, surname, nickname, imageId FROM users WHERE id !=' + userId + ' ORDER BY name ASC');
    for (let i = 0; i < users.length; i++) {
      let chatId = await database.query("SELECT chats.id FROM chats INNER JOIN userChatAssociations ON userChatAssociations.chatId = chats.id INNER JOIN( SELECT userChatAssociations.chatId FROM userChatAssociations WHERE userChatAssociations.userId = " + userId + " )utenteDue ON utenteDue.chatId = userChatAssociations.chatId WHERE userChatAssociations.userId = " + users[i].id + " AND chats.isGroup != 1");
      dataToSand.push({
        id: users[i].id,
        name: users[i].name,
        surname: users[i].surname,
        nickname: users[i].nickname,
        imageId: users[i].imageId,
        chatId: (chatId[0] == undefined ? null : chatId[0].id),
      });
    }
    socket.token = data.token;
    socket.event = "contacts";
    require('../../../app')._CONNECTIONS.add(socket);
    socket.write(JSON.stringify({
      event: 'contacts',
      status: 200,
      data: dataToSand
    }));
  } else {
    socket.write(JSON.stringify({ event: 'contacts', status: 401, error: 'Non sei autorizzato a visionare queste chat!' }));
  }
}